gui/src/app/pages/HomePage/SaveProjectWindow.tsx

```diff
@@ -204,6 +283,52 @@ const GistExportView: FunctionComponent<GistExportViewProps> = ({
   );
 };
 
+type SpecifyGistUrlToUpdateComponentProps = {
+  gistUrl: string | null;
+  setGistUrl: (gistUrl: string) => void;
+};
+
+const SpecifyGistUrlToUpdateComponent: FunctionComponent<
+  SpecifyGistUrlToUpdateComponentProps
+> = ({ gistUrl, setGistUrl }) => {
+  return (
+    <table className="project-summary-table">
+      <tbody>
+        <tr>
+          <td>GitHub Gist URL to update</td>
+          <td>
+            <input
+              type="text"
+              value={gistUrl || ""}
+              onChange={(e) => setGistUrl(e.target.value)}
+            />
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  );
+};
+
+const createPatchForUpdatingGist = (
+  existingFiles: { [key: string]: string },
+  newFiles: { [key: string]: string },
+) => {
+  const patch: { [key: string]: string | null } = {};
+  for (const fname in newFiles) {
+    const newContent = newFiles[fname];
+    if (existingFiles[fname] === newContent) continue;
+    if (!newContent.trim()) continue;
+    patch[fname] = newContent;
+  }
+  // handle deleted files
+  for (const fname in existingFiles) {
+    if (!newFiles[fname] || !newFiles[fname].trim()) {
+      patch[fname] = null;
+    }
+  }
+  return patch;
+};
+
```

Comment:
Gist-manipulation logic should go in the `gists/SaveAsGithubGist` rather than in the UI component

Reply:
Done